apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -74,6 +77,7 @@ type ClusterQueueSpec struct {
 
 	// flavorFungibility defines whether a workload should try the next flavor
 	// before borrowing or preempting in the flavor being evaluated.
+	// +kubebuilder:default={}
```

Comment:
NIT: Just making the field not a pointer will do this and we can remove some nil checks in the rest of the code-base.

Reply:
Should this change be applied to every field where `+kubebuilder:default={}`  was added? Because we could change the pointers on `spec.preemption` and `spec.preemption.borrowWithinCohort` as well

Reply:
we can make it a non-pointer when we migrate to v1beta2 or v1.